src/main/java/org/jruby/ext/openssl/OCSPBasicResponse.java

```diff
@@ -0,0 +1,147 @@
+* DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+* ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+* DEALINGS IN THE SOFTWARE.
+*
+*  Copyright (C) 2007-2009 Ola Bini <ola.bini@gmail.com>
+*  Copyright (C) 2009-2016 The JRuby Team
+*
+* Alternatively, the contents of this file may be used under the terms of
+* either of the GNU General Public License Version 2 or later (the "GPL"),
+* or the GNU Lesser General Public License Version 2.1 or later (the "LGPL"),
+* in which case the provisions of the GPL or the LGPL are applicable instead
+* of those above. If you wish to allow use of your version of this file only
+* under the terms of either the GPL or the LGPL, and not to allow others to
+* use your version of this file under the terms of the EPL, indicate your
+* decision by deleting the provisions above and replace them with the notice
+* and other provisions required by the GPL or the LGPL. If you do not delete
+* the provisions above, a recipient may use your version of this file under
+* the terms of any one of the EPL, the GPL or the LGPL.
+*
+*
+* JRuby-OpenSSL includes software by The Legion of the Bouncy Castle Inc.
+* Please, visit (http://bouncycastle.org/license.html) for licensing details.
+*/
+package org.jruby.ext.openssl;
+
+import java.security.SecureRandom;
+
+import org.bouncycastle.asn1.ocsp.BasicOCSPResponse;
+import org.jruby.Ruby;
+import org.jruby.RubyClass;
+import org.jruby.RubyFixnum;
+import org.jruby.RubyModule;
+import org.jruby.RubyObject;
+import org.jruby.RubyString;
+import org.jruby.anno.JRubyMethod;
+import org.jruby.exceptions.RaiseException;
+import org.jruby.runtime.Arity;
+import org.jruby.runtime.ObjectAllocator;
+import org.jruby.runtime.ThreadContext;
+import org.jruby.runtime.Visibility;
+import org.jruby.runtime.builtin.IRubyObject;
+
+import static org.jruby.ext.openssl.OCSP._OCSP;
+
+/*
+ * An OpenSSL::OCSP::BasicResponse contains the status of a certificate
+ * check which is created from an OpenSSL::OCSP::Request. 
+ * A BasicResponse is more detailed than a Response.
+ * 
+ * @author lampad
+ */
+public class OCSPBasicResponse extends RubyObject {
+    private static final long serialVersionUID = 8755480816625884227L;
+    
+    private static ObjectAllocator BASICRESPONSE_ALLOCATOR = new ObjectAllocator() {
+        public IRubyObject allocate(Ruby runtime, RubyClass klass) {
+            return new OCSPBasicResponse(runtime, klass);
+        }
+    };
+    
+    public static void createBasicResponse(final Ruby runtime, final RubyModule _OCSP) {
+        RubyClass _BasicResponse = _OCSP.defineClassUnder("BasicResponse", runtime.getObject(), BASICRESPONSE_ALLOCATOR);
+        _BasicResponse.defineAnnotatedMethods(OCSPBasicResponse.class);
+    }
+    
+    private RubyString der;
+    private byte[] nonce;
+    private SecureRandom random = new SecureRandom();
```

Comment:
we should delay the `SecureRandom` creation - and use the helper from `SecurityHelper` (this is minor, really)